repo/blob/s3/s3_storage.go

```diff
@@ -133,6 +133,8 @@ func (s *s3Storage) getVersionMetadata(ctx context.Context, b blob.ID, version s
 
 func (s *s3Storage) PutBlob(ctx context.Context, b blob.ID, data blob.Bytes, opts blob.PutOptions) error {
 	switch {
+	case opts.HasRetentionOptions() && !opts.RetentionMode.IsValidS3():
```

Comment:
Same as above, we need to figure out a blob storage agnostic way of doing this.

Reply:
But this is the individual clients so they're already not agnostic e.g. s3 using minio